packages/stark-ui/src/modules/date-picker/components/date-picker.component.html

```diff
@@ -1,17 +1,21 @@
-<mat-form-field [id]="pickerId">
-	<input
-		matInput
-		[disabled]="isDisabled"
-		[id]="pickerId + '-input'"
-		[matDatepicker]="picker"
-		[placeholder]="label | translate"
-		[matDatepickerFilter]="dateFilterFnWrapper"
-		[min]="minDate"
-		[max]="maxDate"
-		[name]="pickerName"
-		[value]="date"
-		(dateChange)="onDateChange($event)"
-	/>
+<input
+	matInput
+	[id]="pickerId + '-input'"
+	[matDatepicker]="picker"
+	[value]="value"
+	(dateInput)="onDateInput($event)"
+	(dateChange)="onDateChange($event)"
+	[matDatepickerFilter]="dateFilterFnWrapper"
+	[min]="min"
+	[disabled]="disabled"
+	[max]="max"
+	[name]="pickerName"
+	[required]="required"
+	(focus)="onFocus($event)"
```

Comment:
I think the `onFocus()` method can be simplified if you pass directly the input element by doing this: 

```html
<input matInput
     #pickerInput="matDatepickerInput"   <---- this will export this input as "pickerInput"
     ...
     (focus)="onFocus(pickerInput)"  <---- so in the onFocus() you already get the input element
     
```

Reply:
Unfortunately, we need to change the HTML value thanks to the target to make it working well. It's a bit tricky but it is working fine. We could take a look on it later and see if Angular proposes something for this.